src/component.js

```diff
@@ -608,7 +613,14 @@ class Component {
  async run(steps, filter, recursive, cntx) {
     // collect steps from descs, interits, parents
     for(const step of steps) {
-      const list2execute = this.findStep(step, filter, this.home, cntx.clone(), [], []);
+      let s = step;
+      let c = null;
+      const parts = step.split('@');
+      if (parts.length > 1) {
+        s = parts[0];
+        c = this.resolve([parts[1]]);
+      }
+      const list2execute = this.findStep(s, filter, this.home, cntx.clone(), [], [], c[0]);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') TypeError can be thrown as "c" might be null or undefined here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://159.65.230.204:9000/coding_rules#rule_key=javascript%3AS2259)